### DISCUSSION TITLE
cannot handle NotOwner error from cog_check

### DISCUSSION THREAD

**Discussion by aurkaxi** [Rapptz/discord.py#9037] [General] [answered]
my error handling:
```py
   @commands.Cog.listener()
    async def on_command_error(self, ctx, error):
        if isinstance(error, commands.NotOwner):
            await ctx.send(f"Not owner.")
        elif isinstance(error, commands.CheckFailure):
            await ctx.send(f"Check failure.")
```
my cog check:
```py
class adminusage(commands.Cog, name="Admin Usage"):
    def __init__(self, bot):
        self.bot = bot

    async def cog_check(self, ctx):
        return await self.bot.is_owner(ctx.author)
```

so when the check fails it sends does the job for CheckFailure. But it should send NotOwner.

(CheckFailure always denies to work with me :( is this because i am ugly?)

**Comment by LostLuma** [comment, marked as answer]
Hello, this is because the [`is_owner`](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html#discord.ext.commands.Bot.is_owner) method only returns a bool, rather than an exception like the [built in `is_owner` check](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html#discord.ext.commands.is_owner).

You can modify your `cog_check` to instead of simply returning a bool, which will make the library raise `CheckFailure`, raise the `NotOwner` error yourself to handle it elsewhere.

**Reply by Soheab** [reply]
You can also manually call the decorator with the following:
```py
ret = await commands.is_owner().predicate(ctx)
```
that will return a bool or raise `commands.NotOwner` like normal.

Do be aware that the above method is not documented or recommended anywhere.

**Reply by aurkaxi** [reply]
thanks @Soheab ! that is exactly i was looking for hours. I didn't want to raise my own error for something builtin.
